.github/workflows/lintcheck.yml

```diff
@@ -37,57 +34,67 @@ jobs:
         rm -rf lintcheck
         git checkout ${{ github.sha }} -- lintcheck
 
+    - name: Cache lintcheck bin
+      id: cache-lintcheck-bin
+      uses: actions/cache@v4
+      with:
+        path: target/debug/lintcheck
+        key: lintcheck-bin-${{ hashfiles('lintcheck/**') }}
```

Comment:
This might fail, when we update our toolchain. Can you push a commit, where you update the `rust-toolchain` file by a day (this hopefully doesn't break anything else due to rustc changes)? If that breaks this caching action, we might also need to put the rust version in this hash.

Reply:
https://github.com/rust-lang/rust-clippy/actions/runs/9645498686

The lintcheck part worked, it should be fine for the lintcheck bin to be built by whatever toolchain as it doesn't use any rustc internals

Diagnostics pointing to std source changing makes sense, might be able to trim those paths also though I'm not sure why it's in `/rustc/HASH` instead of the toolchains dir

Reply:
Thanks for checking! 

Trimming those paths would be good I think. Otherwise, this might be shown in every PR that is not based on the most recent sync.